src/test/scala/com/microsoft/hyperspace/index/HyperspaceSuite.scala

```diff
@@ -72,4 +72,14 @@ trait HyperspaceSuite extends SparkFunSuite with SparkInvolvedSuite {
       }
     }
   }
+
+  protected def withSparkConf(confName: String, confValue: Any)(f: => Unit): Unit = {
```

Comment:
Might be worth adding a comment that this isn't thread-safe?

Reply:
We can add it, but looks like for testing, Spark doesn't explicitly mention this. ([e.g. here](https://github.com/apache/spark/blob/7955b3962ac46b89564e0613db7bea98a1478bf2/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/plans/PlanTest.scala#L182)))